### PR TITLE
minor changes in data collection in Feature.jsm, added TELEMETRY.md and schema.json (fixes #62)

### LINKED ISSUE #62: Make Telemetry pings SQL-friendly. Write TELEMETRY.md.
#63 Should be closed before working on this issue.

### What data do we want to obtain from pings?
What all do we want to know from these pings? Values are on average per user per branch.
* Does the user engage actively with the introduction panel? (button click versus passive dismiss)
* Does the user engage actively with the pageAction panel? (button click versus passive dismiss)
* What is the value of the badge when the user clicks the pageAction button?
* How many times is the pageAction panel opened?
* How long is the pageAction panel opened? Intro panel?
* How many times is the `about:newtab` page opened?
* How long is the `about:newtab` page opened for?
* If the user ends the study due to: being ineligible (passive), addon disable, modifying built-in tracking protection, opting out of the study via the panel UI.

@gregglind, @MattGrimes, am I missing any important data in telemetry for this study? I will send pdol this list once it is finalized.

### Ensure pings are SQL-friendly
A key in the object passed into the telemetry should be `category`. You may need 2-3 fields to help more easily sort pings. Ensure there is no transformation necessary of the string; that each string key and string value can be used as-is.

Examples:
* Group by `"event"` key where the value is `"panel-dismissed"`.
* Each panel has a set of events with values like: `"open"`, `"interact"`, `"close"`.
### TELEMETRY.md
This document is intended to list and describe all the events and conditions leading to that telemetry event. Also list all the things that can be known from the data collected (ex: number of times the user clicked the pageAction button as a measure of engagement).

### PR DESCRIPTION
**Pull request by raymak**

### CODE REVIEW

File: TELEMETRY.md [hunk 1]
```diff
@@ -1,85 +1,332 @@
...
 ## `shield-study` pings (common to all shield-studies)
 
 `shield-studies-addon-utils` sends the usual packets.
 
 The STUDY SPECIFIC ENDINGS this study supports are:
 
-- "voted",
-- "notification-x"
-- "window-or-fx-closed"
+- "expired"
```
Comment (biancadanforth) [inline TELEMETRY.md]: Nit here, but isn't "expired" a standard i.e. `shield-study` ending?

File: TELEMETRY.md [hunk 2]
```diff
@@ -1,85 +1,332 @@
...
 These are the `payload` fields from all pings in the `shield-study` and `shield-study-addon` buckets.
 
 ```
 
 // common fields
 
-branch        up-to-expectations-1        // should describe Question text
-study_name    57-perception-shield-study
-addon_version 1.0.0
+branch        private        // should describe Question text
```
Comment (biancadanforth) [inline TELEMETRY.md]: Nit: remove comment from template?

File: TELEMETRY.md [hunk 3]
```diff
@@ -1,85 +1,332 @@
...
+study_name    trackingProtectionMessagingExperiment
```
Comment (biancadanforth) [inline TELEMETRY.md]: This `study_name` will actually be one of: addon-tracking-protection-messaging-1433473-new-users or addon-tracking-protection-messaging-1433473-all-users per #157 . If you could update it, I'll make sure it gets merged today.

File: addon/lib/Feature.jsm [hunk 4]
```diff
@@ -912,7 +912,7 @@ class Feature {
 
   async reportBehaviorSummary() {
     const behaviorSummary = await Storage.get("behavior-summary");
-    await this.telemetry(Object.assign({message_type: "behavior_summary"}, behaviorSummary));
+    await this.telemetry(Object.assign({message_type: "behavior-summary"}, behaviorSummary));
```
Comment (biancadanforth) [inline addon/lib/Feature.jsm]: Good catch

File: schemas/schema.json [hunk 5]
```diff
@@ -0,0 +1,225 @@
+{
```
Comment (biancadanforth) [inline schemas/schema.json]: I know what schemas are used for generally, but when would this schema be used? I don't see us validating the structure of pings before sending them.

File: TELEMETRY.md (hunk 1 above)
Comment (raymak) [reply TELEMETRY.md]: True

File: schemas/schema.json (hunk 5 above)
Comment (raymak) [reply schemas/schema.json]: It's not just for validating. It helps the analyst to know what exactly is contained in each ping and each field.

**Pull request merged**